File: 2021/python/day2.py
from utils import getInputData, getSpiltList

data = getInputData(year=2021)
data = getSpiltList(data, " ")

horizontal = 0
depth = 0

for instruction in data:
    direction = instruction[0]
    instruction[1] = int(instruction[1])
    value = instruction[1]
    if direction == "forward":
        horizontal += value
    elif direction == "down":
        depth += value
    elif direction == "up":
        depth -= value

print("Part one: ", horizontal * depth)

horizontal = 0
depth = 0
aim = 0

for instruction in data:
    direction = instruction[0]
    value = instruction[1]
    if direction == "forward":
        horizontal += value
        depth += value * aim

    elif direction == "down":
        aim += value
    elif direction == "up":
        aim -= value


print("Part two: ", horizontal * depth)

File: 2021/golang/day2.go
package main

import (
	"log"
	"strconv"
)

func day2() {

	data := getSplitList(getInputData(2021, 2), " ")
	day2part1(data)
	day2part2(data)

}

func day2part1(data [][]string) {

	var horizontal, depth int

	for _, instruction := range data {
		direction := instruction[0]
		value, err := strconv.Atoi(instruction[1])
		if err != nil {
			log.Fatal(err)
		}

		switch direction {
		case "forward":
			horizontal += value
		case "down":
			depth += value
		case "up":
			depth -= value
		}
	}

	log.Println("Answer for Day 2, Part 1 is:", horizontal*depth)

}

func day2part2(data [][]string) {
	var horizontal, depth, aim int

	for _, instruction := range data {
		direction := instruction[0]
		value, err := strconv.Atoi(instruction[1])
		if err != nil {
			log.Fatal(err)
		}

		switch direction {
		case "forward":
			horizontal += value
			depth += value * aim
		case "down":
			aim += value

		case "up":
			aim -= value
		}
	}

	log.Println("Answer for Day 2, Part 2 is:", horizontal*depth)
}
